api/v1alpha1/limitador_types.go

```diff
@@ -61,6 +62,9 @@ type LimitadorSpec struct {
 
 	// +optional
 	Limits []RateLimit `json:"limits,omitempty"`
+
+	// +optional
+	PodDisruptionBudget *policyv1.PodDisruptionBudgetSpec `json:"pdb,omitempty"`
```

Comment:
If the user defines the `Selector` part of the spec the operator ignores that part of the spec. I think there is two ways to of doing this. We reconcile the selectors if the user specifies them or we don't allow the user to specify the selector in which case we cant use `policyv1.PodDisruptionBubgetSpec` as the resource type.

The #75 does not state which would be the preferred choice but I would assume we would not want the user stating the selector.

Reply:
Yep I agree that it is a little confusing. The selector should not be set realistically as the limitador-operator owns the deployment with the labels that the PDB should be selecting. On the other hand does it make sense to create a new PDB Spec type bespoke to limitador to exclude this field and then map it back to the existing one? this definitely makes the API clearer but makes us (nearly) duplicate an existing type - curious on further thoughts here

Reply:
I think it makes more sense to make our own custom stuct with just the available fields that is allowed by the user even if it's duplicating the PDB spec if we don't allow them to specifiy the `Selector` field.

The reason for this is it prevents user confusion from the API provided by this operator. By using the full PDB spec in our CRD, the PDB spec descriptions will be added to our CRDs and will give an expected impression that the `Selector` field is supported in the spec when in fact it does not 🤔 

Unless we are supporting the `Selector` field, I'm in favour of using our own type without this field. In the future, if it does become supported, we can use the PBD Spec instead

Reply:
Sure the more I think about this the more sense it makes - I'll make a change to include our own PDB spec